### PR TITLE
Ignore jobs if integration is disabled

### LINKED ISSUE #877: Not allowed create job if integration is disabled
**What happened**:
When we apply any job from an integration (JobSet, MPIJob, etc) with a local queue specified in the metadata.labels section and this integration is disabled in the queue controller configuration, we can see the uncontrolled behavior of creating / deleting jobs.

**What you expected to happen**:
When applying a job with ```kueue.x-k8s.io/queue-name``` label, get an error saying that integration is not enabled

**How to reproduce it (as minimally and precisely as possible)**:

1. Deploy Kueue controller manager with default configuration
2. Deploy Jobset controller manager
3. Create local queue
4. Apply one of JobSet examples with defined ```kueue.x-k8s.io/queue-name``` label

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by stuton**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
This fix ignores jobs if the integration of these jobs is disabled, but there is ```kueue.x-k8s.io/queue-name``` label in the metadata of jobs

#### Which issue(s) this PR fixes:

Fixes #877

#### Does this PR introduce a user-facing change?

```release-note
Ignoring integrations if crd wasn't installed otherwise all integrations are enabled by default
```

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
Enable everything by default here https://github.com/kubernetes-sigs/kueue/blob/main/config/components/manager/controller_manager_config.yaml and in the helm chart.

**Comment by alculquicondor** [comment]
This requires a release note.

File: main.go [hunk 1]
```diff
@@ -205,24 +210,28 @@ func setupControllers(mgr ctrl.Manager, cCache *cache.Cache, queues *queue.Manag
...
-				mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, constants.KueueName)),
-				opts...,
-			).SetupWithManager(mgr); err != nil {
-				log.Error(err, "unable to create controller")
-				return err
-			}
-			if err := cb.SetupWebhook(mgr, opts...); err != nil {
-				log.Error(err, "Unable to create webhook")
-				return err
+			if crds.Has(name) {
```
[the remark this answers is not in the document]
Comment (stuton) [reply main.go:222]: It's added here by default:
https://github.com/kubernetes-sigs/kueue/pull/883/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R359

**Comment by alculquicondor** [comment]
Awesome. Did you do a manual test with MPIJob?

Could you add the release note?

**Comment by stuton** [comment]
I've just run it again to make sure it works. Sure, release note was updated

**Pull request merged**